src/hotspot/share/logging/logConfiguration.cpp

```diff
@@ -262,6 +262,9 @@ void LogConfiguration::configure_output(size_t idx, const LogSelectionList& sele
   }
 
   if (!enabled && idx > 1) {
+    // User may disable a logOuput like this:
+    // LogConfiguration::parse_log_arguments(filename, "all=off", "", "", &stream);
+    // Just be conservative. Flush them all before deleting idx.
```

Comment:
Thanks for the comment. Note that this is still a best-effort attempt because immediately after the flush completes further logging could happen before the idx is deleted. So the async writer thread can still encounter  a fatal error. But as this can only happen with direct internal/gtest use of the LogConfiguration API I don't think this is a concern. Had I understood the issue sooner (my fault) I would have suggested just fixing the test to disallow async mode rather than making a change to the functional code. But this is something we can re-examine later.

Reply:
Note: my comment about further logging being possible is because I can't obviously tell that something prior to this would have caused it to be disabled. But that may be the case, and then there is no issue with the flush at  all.

Reply:
Thanks for the reviewing.  Theoretically, it's possible to enqueue a log message after flush(). 
That's a race condition and even original synchronous logging can't handle it well.  We don't find code to do that. 

This output deletion is a classic safe memory reclamation. There's still room to improve it.